pkg/api/handlers/compat/images.go

```diff
@@ -404,26 +405,44 @@ func GetImage(w http.ResponseWriter, r *http.Request) {
 }
 
 func GetImages(w http.ResponseWriter, r *http.Request) {
-	images, err := utils.GetImages(w, r)
+	decoder := r.Context().Value(api.DecoderKey).(*schema.Decoder)
+	runtime := r.Context().Value(api.RuntimeKey).(*libpod.Runtime)
+	query := struct {
+		All     bool
+		Digests bool
+		Filter  string // Docker 1.24 compatibility
+	}{
+		// This is where you can override the golang default value for one of fields
+	}
+
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		utils.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest,
+			errors.Wrapf(err, "failed to parse parameters for %s", r.URL.String()))
+		return
+	}
+	if _, found := r.URL.Query()["digests"]; found && query.Digests {
+		utils.UnSupportedParameter("digests")
+		return
+	}
+
+	filterList, err := filters.FiltersFromRequest(r)
 	if err != nil {
-		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "Failed get images"))
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, err)
 		return
 	}
+	if !utils.IsLibpodRequest(r) && len(query.Filter) > 0 { // Docker 1.24 compatibility
+		filterList = append(filterList, "reference="+query.Filter)
+	}
 
-	summaries := make([]*entities.ImageSummary, 0, len(images))
-	for _, img := range images {
-		// If the image is a manifest list, extract as much as we can.
-		if isML, _ := img.IsManifestList(r.Context()); isML {
-			continue
-		}
+	imageEngine := abi.ImageEngine{Libpod: runtime}
 
-		is, err := handlers.ImageToImageSummary(img)
```

Comment:
By skipping the call to `ImageToImageSummary`:
* `ImageSummary.VirtualSize` will be incorrect.
* It appears `ID` will no longer be prefixed with `sha256:`
* `ImageSummary.RepoDigests` will no longer be formatted  rather the raw data will be returned.
* ...

Reply:
I have fixed the sha256.
VirtualSize seems to be the same between the two calls. (It is always just size).
I switched libpod output for RepoDigests to match Dockers.
I did not see any others.